src/plugins/StateService.js

```diff
@@ -14,7 +14,9 @@ const StateService = {
         situation || this.$store.state.situation,
         this.$router
       )
-      this.$router.push(nextStep.path)
+      this.$router.push(nextStep.path).catch((failure) => {
+        throw new Error(failure)
+      })
```

Comment:
En quoi cela est-il différent de la version précédente ?

Reply:
L'hypothèse que je souhaite valider, [à partir de ce thread d'un développeur de Vue](https://github.com/vuejs/vue-router/issues/2881#issuecomment-520554378), c'est que les erreurs Sentry qui indiquent que la navigation a été "cancelled" est causé par la promesse qui n'est pas résolu dans la version du router de vue en version 3.1  Depuis [vue-router 3.4](https://router.vuejs.org/guide/advanced/navigation-failures.html#detecting-navigation-failures), et l'ajout d'une promesse au `this.router` , le catch failure permet d'éviter les erreurs de ce type. Quant au throw new error, je l'ai ajouté parce que je pensais qu'on ne voulait pas ignorer ce type d'erreur.


Plus d'informations : 
https://github.com/vuejs/vue-router/issues/2881#issuecomment-520554378
https://stackoverflow.com/questions/62223195/vue-router-uncaught-in-promise-error-redirected-from-login-to-via-a